### PR TITLE
Readme update

### LINKED ISSUE #214: Update README.md
https://github.com/web-push-libs/web-push/pull/198

### PR DESCRIPTION
**Pull request by gauntface**
@marco-c 

- Updates the readme for the new API.
- Adds browser support info.
- Updates dependencies.
- Adds info on using VAPID key in subscribe() call.

Fixes #214.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,74 +1,372 @@
...
+
+    const vapidPublicKey = '<Your Public Key from generateVAPIDKeys()>';
+    const convertedVapidKey = urlBase64ToUint8Array(vapidPublicKey);
+
+    registration.pushManager.subscribe({
+      userVisibleOnly: true,
+      applicationServerKey: convertedVapidKey  
+    });
+
+# API Referance
```
Comment (marco-c) [inline README.md]: `Reference`

File: README.md [hunk 2]
```diff
@@ -1,74 +1,372 @@
...
-- *privateKey*;
-- *publicKey*.
+    npm install web-push --save
+
+# Usage
+
+The common use case for this library is an application server using
+a GCM API key and VAPID keys.
+
+    const webpush = require('web-push');
```
Comment (marco-c) [inline README.md]: Can we enable syntax highlighting for the code snippets?

File: README.md [hunk 3]
```diff
@@ -1,74 +1,372 @@
...
+    const convertedVapidKey = urlBase64ToUint8Array(vapidPublicKey);
+
+    registration.pushManager.subscribe({
+      userVisibleOnly: true,
+      applicationServerKey: convertedVapidKey  
+    });
+
+# API Referance
+
+## sendNotification(pushSubscription, payload, options)
```
Comment (marco-c) [inline README.md]: Can we somehow specify that `sendNotification` can also be used without sending a payload and/or without using VAPID?

Comment (gauntface) [inline README.md]: I've add a note regarding payload and said that the method will work if the push service doesn't require GCM API and / or VAPID. Happy to change further.

**Comment by gauntface** [comment]
@marco-c Travis was failing due to eslint dropping support for older versions of node. Might be time for web-push to do the same ;)

**Comment by marco-c** [comment]
> @marco-c Travis was failing due to eslint dropping support for older versions of node. Might be time for web-push to do the same ;)

Yeah, I hear you. Perhaps we can do that now and make people with older Node.js versions stick to version 2.2.0 of the library.

I see `fxa-auth-server` (which was my main concern) switched to the latest Node.js version in https://github.com/mozilla/fxa-auth-server/commit/f129b7b56170e2114da0421d42f83dc6664a3224 and https://github.com/mozilla/fxa-auth-server/pull/1330.

**Comment by marco-c** [comment]
I filed #225 for removing support for older Node.js versions.

**Pull request merged**